src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,135 @@
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setPassword(null);
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException expected for user.password = null");
+    }
+
+    @Test
+    void register_loginAlreadyInStorage_notOk() {
+        User storageUser = new User();
+        storageUser.setLogin("second_login");
+        storageDao.add(storageUser);
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin("second_login");
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException, method allows user duplicates");
+    }
+
+    @Test
+    void register_loginIsEmpty_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin("");
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException,"
+                + "when login field empty an exception should be thrown");
+    }
+
+    @Test
+    void register_default_ok() {
+        registrationService.register(defaultUser);
+        assertEquals(storageDao.get(defaultUser.getLogin()).getLogin(), defaultUser.getLogin());
+        assertEquals(storageDao.get(defaultUser.getLogin()).getPassword(),
+                defaultUser.getPassword());
+        assertEquals(storageDao.get(defaultUser.getLogin()).getAge(), defaultUser.getAge());
+    }
+
+    @Test
+    void register_passwordLengthLessThan6_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin("third_login");
+            defaultUser.setPassword("");
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException,"
+                + "if password length less than 6 an exception should be thrown");
+    }
+
+    @Test
+    void register_AgeLessThan18_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin("fourth_login");
+            defaultUser.setAge(16);
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException, if age < 18 method you should thrown an exception");
+    }
+
+    @Test
+    void register_AgeEquals18_ok() {
+        defaultUser.setAge(18);
+        defaultUser.setLogin("login_fifth");
+        registrationService.register(defaultUser);
+        assertEquals(storageDao.get(defaultUser.getLogin()).getAge(),
+                defaultUser.getAge(), "Age not match");
+    }
+
+    @Test
+    void register_AgeIsNegative_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin("sixth_login");
+            defaultUser.setAge(-23939);
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException, if age < 0 method should thrown an exception");
+    }
```

Comment:
we should separate the test data preparation from the actual test to improve readability, follow this example for other similar cases as well:
```suggestion
    @Test
    void register_AgeIsNegative_notOk() {
        defaultUser.setAge(-23939);
        
        assertThrows(InvalidRegistrationDataException.class, () -> {
            registrationService.register(defaultUser);
        }, "InvalidRegistrationDataException, if age < 0 method should thrown an exception");
    }
```